dndoptions.clo

```diff
@@ -0,0 +1,95 @@
+\ExplSyntaxOn
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% Class / Package options
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+% Provides class / package option processing of l3keys in LaTeX2e
+\RequirePackage {l3keys2e}
+
+% deprecate warning used in options and throughout the collection
+\msg_new:nnn {dnd} {deprecated} { #1 ~ is ~ deprecated ~ and ~ will ~ be ~ removed ~ in ~ version ~ #2. ~ #3 }
+
+\cs_new_protected:Npn { \__dnd_deprecated:nnn } #1#2#3
+  {
+    \msg_warning:nnnnn { dnd } { deprecated } {#1} {#2} {#3}
+  }
+
+\cs_new_protected:Npn { \__dnd_deprecated:nn } #1#2
+  {
+    \msg_warning:nnnnn { dnd } { deprecated } {#1} {#2} {}
+  }
+
+% These booleans are not set directly by an option (i.e. instead set by .code)
+\bool_new:N \l__dnd_show_background_bool
+\bool_new:N \l__dnd_show_footer_scroll_bool
+\bool_new:N \l__dnd_justified_bool
+\bool_new:N \l__dnd_multitoc_bool
+\bool_new:N \l__dnd_layout_bool
```

Comment:
`\l__dnd_layout_bool` should default to `true` for now. We should preserve the current `dnd` package behaviour at least until 1.0.

We can add a deprecation warning for using `layout=true` with package `dnd`.

Reply:
The default is now true for packages, and it emits a warning if the layout flag was not set while using the package. There are also some comments to make obvious what to roll back in 1.0.0.